pipeline/splicing/calc_priors/dataproc.py

```diff
@@ -0,0 +1,144 @@
+import collections
+
+
+# idea from https://stackoverflow.com/a/28452633/346905
+from calc_priors.compute import getVarLocation
+from calc_priors.extract import getVarType
+from calc_priors.verify import getVarStrand
+
+
+class ReadOnlyDict(collections.Mapping):
+    def __init__(self, data):
+        self._data = data
+
+    def __getitem__(self, key):
+        return self._data[key]
+
+    def __len__(self):
+        return len(self._data)
+
+    def __iter__(self):
+        return iter(self._data)
+
+
+# FIXME: make this read-only again and update addVarDataToRow to base its copy off this
```

Comment:
is this a must fix or can it be addressed in a followup? i.e. will it cause errors?

Reply:
It's a "nice-to-have"; it'll work as it is.

Now that I'm thinking about this again, rather than worry about making the dict read-only it'd be better to add a class to represent what we're currently piling into BLANK_DICT, but that can be done in a followup.